<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from src/raido-bot_description/urdf/raido-bot.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="raido-bot">
  <material name="grey">
    <color rgba="0.38 0.38 0.38 1.0"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1.0"/>
  </material>
  <link name="base_link">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="5.0"/>
      <inertia ixx="0.043125" ixy="0.0" ixz="0.0" iyy="0.0890625" iyz="0.0" izz="0.1040625"/>
    </inertial>
    <collision name="base_link_col">
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.5 0.35 0.11"/>
      </geometry>
    </collision>
    <visual name="base_link_vis">
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://raido-bot_description/meshes/base_link.stl" scale="1 1 1"/>
      </geometry>
      <material name="grey"/>
    </visual>
  </link>
  <link name="wheel_link_fr">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="1.0"/>
      <inertia ixx="0.0025" ixy="0.0" ixz="0.0" iyy="0.0025" iyz="0.0" izz="0.001"/>
    </inertial>
    <collision name="wheel_link_fr_col">
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <cylinder length="0.05" radius="0.06"/>
      </geometry>
    </collision>
    <visual name="wheel_link_fr_vis">
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://raido-bot_description/meshes/wheel_link.stl" scale="1 1 1"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="wheel_link_fr_joint" type="continuous">
    <origin rpy="0.0 0.0 0.0" xyz="0.15 0.19999999999999998 0.0"/>
    <parent link="base_link"/>
    <child link="wheel_link_fr"/>
    <axis xyz="0 1 0"/>
    <limit effort="1.0" lower="0.0" upper="0.0" velocity="1.0"/>
  </joint>
  <link name="wheel_link_br">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="1.0"/>
      <inertia ixx="0.0025" ixy="0.0" ixz="0.0" iyy="0.0025" iyz="0.0" izz="0.001"/>
    </inertial>
    <collision name="wheel_link_br_col">
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <cylinder length="0.05" radius="0.06"/>
      </geometry>
    </collision>
    <visual name="wheel_link_br_vis">
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://raido-bot_description/meshes/wheel_link.stl" scale="1 1 1"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="wheel_link_br_joint" type="continuous">
    <origin rpy="0.0 0.0 0.0" xyz="-0.15 0.19999999999999998 0.0"/>
    <parent link="base_link"/>
    <child link="wheel_link_br"/>
    <axis xyz="0 1 0"/>
    <limit effort="1.0" lower="0.0" upper="0.0" velocity="1.0"/>
  </joint>
  <link name="wheel_link_fl">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="1.0"/>
      <inertia ixx="0.0025" ixy="0.0" ixz="0.0" iyy="0.0025" iyz="0.0" izz="0.001"/>
    </inertial>
    <collision name="wheel_link_fl_col">
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <cylinder length="0.05" radius="0.06"/>
      </geometry>
    </collision>
    <visual name="wheel_link_fl_vis">
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://raido-bot_description/meshes/wheel_link.stl" scale="1 1 1"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="wheel_link_fl_joint" type="continuous">
    <origin rpy="0.0 0.0 0.0" xyz="0.15 -0.19999999999999998 0.0"/>
    <parent link="base_link"/>
    <child link="wheel_link_fl"/>
    <axis xyz="0 1 0"/>
    <limit effort="1.0" lower="0.0" upper="0.0" velocity="1.0"/>
  </joint>
  <link name="wheel_link_bl">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="1.0"/>
      <inertia ixx="0.0025" ixy="0.0" ixz="0.0" iyy="0.0025" iyz="0.0" izz="0.001"/>
    </inertial>
    <collision name="wheel_link_bl_col">
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <cylinder length="0.05" radius="0.06"/>
      </geometry>
    </collision>
    <visual name="wheel_link_bl_vis">
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://raido-bot_description/meshes/wheel_link.stl" scale="1 1 1"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="wheel_link_bl_joint" type="continuous">
    <origin rpy="0.0 0.0 0.0" xyz="-0.15 -0.19999999999999998 0.0"/>
    <parent link="base_link"/>
    <child link="wheel_link_bl"/>
    <axis xyz="0 1 0"/>
    <limit effort="1.0" lower="0.0" upper="0.0" velocity="1.0"/>
  </joint>
  <gazebo>
    <plugin filename="ignition-gazebo-diff-drive-system" name="gz::sim::systems::DiffDrive">
      <left_joint>wheel_link_fl_joint</left_joint>
      <right_joint>wheel_link_fr_joint</right_joint>
      <wheel_separation>0.35</wheel_separation>
      <wheel_radius>0.06</wheel_radius>
      <odom_publish_frequency>0.1</odom_publish_frequency>
      <max_linear_acceleration>1</max_linear_acceleration>
      <min_linear_acceleration>-1</min_linear_acceleration>
      <max_angular_acceleration>2</max_angular_acceleration>
      <min_angular_acceleration>-2</min_angular_acceleration>
      <max_linear_velocity>0.5</max_linear_velocity>
      <min_linear_velocity>-0.5</min_linear_velocity>
      <max_angular_velocity>1</max_angular_velocity>
      <min_angular_velocity>-1</min_angular_velocity>
    </plugin>
  </gazebo>
</robot>
